content/articles/caching-logging-typography-in-flutter/index.md

```diff
@@ -0,0 +1,759 @@
+
+In a situation where there is an HTTP request made to a server for a stale resource (data that does not change) when a user switches between screens in your application, caching is vital in such situations, which improves user experience.
+
+Users do not have to wait for data to be fetched every time they switch between the screens. For data that may not be stable, you can give the user the option to refresh the data fetched and update the cached records if new data is present.
+
+### Benefits of caching
+- Caching saves resources.
+- Caching gives users a better experience.
+- Caching makes your application work faster and better.
+- It helps you create efficient applications.
+
+### Caching implementation
+To show a sample of this, we will perform an HTTP request in our flutter application to fetch data when the user navigates to a new screen where its contents are fetched.
+
+#### Step 1
+In your `main.dart` file set up your flutter application to display a button that navigates to a new screen when clicked.
+
+```dart
+import 'package:flutter/material.dart';
+void main() {
+ runApp(const MyApp());
+}
+class MyApp extends StatelessWidget {
+ const MyApp({Key? key}) : super(key: key);
+  @override
+ Widget build(BuildContext context) {
+   return MaterialApp(
+     title: 'Caching',
+     theme: ThemeData(
+
+       primarySwatch: Colors.blue,
+     ),
+     Home: Scaffold(
+ appBar: AppBar(
+
+   title: Text('Caching'),
+ ),
+ body: CountriesSelect(),
+       );
+   );
+ }
+}
+```
+
+We have a basic flutter application setup with an `app bar` with the title caching in the above snippet. It has a `body` and a `container` that has nothing within it.
+
+```dart
+class _CountriesSelectState extends State<CountriesSelect> {
+ final country = 'USA';
+
+ @override
+ Widget build(BuildContext context) {
+   return ListTile(
+     leading: const Icon(Icons.gps_fixed),
+     title: Text(country),
+     onTap: () {
+       Navigator.push(
+           context,
+           MaterialPageRoute(
+               builder: (context) =>
+                   FetchDataScreen(country: country)));
+     },
+   );
+ }
+}
+```
+
+#### Step 2
+Next, we will make a widget containing a text titled, USA, the country we want to get more information.
+
+```dart
+    onTap: (){
+    Navigator.push(context,
+        MaterialPageRoute(builder: (context) => FetchDataScreen(country: country)));
+    },
+
+```
+
+We specify the screen we wish to navigate to by passing the selected country as a parameter to the named constructor of the `FetchDataScreen` class and navigate there whenever the list is clicked with the help of the `onTap` function property.
+
+Based on the information gotten from the constructor, we store that in a data field, making it accessible to the class.
+
+#### Step 3
```

Comment:
Explain the title properly i.e Step three: Adding this and this in our application